### DISCUSSION TITLE
EC.any_of equivalent in Selenium-Driverless

### DISCUSSION THREAD

**Discussion by oculunto** [kaliiiiiiiiii/Selenium-Driverless#218] [Q&A] [answered]
Hello ! 
I'm trying to recode a selenium / undetected script I had to Selenium-Driverless, but I can't seem to find an equivalent to 
`wait = WebDriverWait(browser, 3) ` and `wait.until(EC.any_of(EC.visibility_of_element_located('xpath',"something"),EC.visibility_of_element_located('xpath',"something else"))` to avoid doing two or more consecutive explicit .find_element() that would take more time.
For only one wait I'm using .find_element() with a timeout instead of wait.until(EC , but i would like to evaluate multiple conditions, is there a way to do it ? 
Appreciate any help !

**Comment by ttlns** [comment]
`.find_element(..., timeout=10)`

**Comment by oculunto** [comment]
Thank you for your answer and for your work! 
 I am using .find_element() with a timeout already, my problem is wanting to stop the wait with many selectors at the same time (I can do an OR expression in XPATH but not with CSS selector for example).
 Here is an example: 
 `weCanGoOn = WebDriverWait(browser, 8).until(EC.any_of(EC.visibility_of_element_located(( 'css selector', 'circle')),EC.visibility_of_element_located(('xpath', "//*[@id='turnstile-wrapper']")),EC.visibility_of_element_located(( 'css selector',"square"))))`
 
 Here as soon as one of these elements is present, it can go on with the code.

**Reply by ttlns** [reply, marked as answer]
just use `asyncio.wait(` with `asyncio.ensure_future` then.
You can lookup the documentation for that
